features/device-config/persistence/api/src/main/java/org/opennms/features/deviceconfig/persistence/api/DeviceConfig.java

```diff
@@ -208,13 +217,35 @@ public boolean equals(Object o) {
                Objects.equals(failureReason, that.failureReason) &&
                Objects.equals(ipInterface, that.ipInterface) &&
                Objects.equals(serviceName, that.serviceName) &&
+               Objects.equals(status, that.status) &&
                Arrays.equals(config, that.config);
     }
 
     @Override
     public int hashCode() {
-        int result = Objects.hash(ipInterface, serviceName, encoding, configType, fileName, failureReason, createdTime, lastUpdated, lastFailed, lastSucceeded);
+        int result = Objects.hash(ipInterface, serviceName, encoding, configType, fileName, failureReason, createdTime, lastUpdated, lastFailed, lastSucceeded, status);
         result = 31 * result + Arrays.hashCode(config);
         return result;
     }
+
+    public static DeviceConfigStatus determineBackupStatus(DeviceConfig dc) {
+        return determineBackupStatus(dc.getLastUpdated(), dc.getLastSucceeded());
+    }
+
+    public static DeviceConfigStatus determineBackupStatus(Date lastUpdated, Date lastSucceeded) {
```

Comment:
We should also update status in `updateDeviceConfigContent` and `updateDeviceConfigFailure`

Reply:
Good catch, fixed